test/integration/query/additional.js

```diff
@@ -1051,6 +1051,68 @@ module.exports = function (knex) {
       });
     });
 
+    it('.timeout(ms, {cancel: true}) should not cancel subsequent queries in a transaction if timeouted query completes while being cancelled', async function () {
```

Comment:
This test will fail without `connectionToKill.pause()`/`resume()` in `Client_MySQL.cancelQuery()` method. The test simulates a race-condition when the timeouted query manages to complete while we are trying to kill it. It is simulated by slowing down `Client_MySQL.acquireRawConnection()` method.

The current behaviour without `connectionToKill.pause()`/`resume()`:

![current](https://user-images.githubusercontent.com/5547275/115765231-cb8f5900-a3a6-11eb-8baa-c0b9d573afbb.jpg)

Fixed behaviour without the race condition with `connectionToKill.pause()`/`resume()`:

![expected](https://user-images.githubusercontent.com/5547275/115765423-06918c80-a3a7-11eb-8e60-0e0ecfcd938b.jpg)